src/mixer/main.go

```diff
@@ -224,6 +226,69 @@ func cmdAddBundles(args []string) {
 	b.AddBundles(bundles, *force, *git)
 }
 
+func cmdInit(args []string) {
+	initcmd := flag.NewFlagSet("init", flag.ExitOnError)
+	confOnlyflag := initcmd.Bool("config-only", false, "Only initialize the builder config")
+	allflag := initcmd.Bool("all", false, "Create a mix with all Clear bundles included")
+	clearflag := initcmd.Int("clearver", 0, "Supply the Clear version to compose the mix from")
+	mixflag := initcmd.Int("mixver", 0, "Supply the Mix version to build")
+	initcmd.Parse(args)
+
+	pwd, err := os.Getwd()
+	if err != nil {
+		helpers.PrintError(err)
+		os.Exit(1)
+	}
+
+	err = os.Mkdir(pwd+"/rpms", 0755)
+	if os.IsExist(err) {
+		fmt.Println("Directory '/rpms' already exists. Skipping...")
+	} else if err != nil {
+		helpers.PrintError(err)
+		os.Exit(1)
+	}
+
+	err = os.Mkdir(pwd+"/local", 0755)
+	if os.IsExist(err) {
+		fmt.Println("Directory '/local' already exists. Skipping...")
+	} else if err != nil {
+		helpers.PrintError(err)
+		os.Exit(1)
+	}
+
+	err = helpers.CopyFile(pwd+"/builder.conf", "/usr/share/defaults/bundle-chroot-builder/builder.conf", false)
+	if os.IsExist(err) {
+		fmt.Println("File 'builder.conf' already exists. Skipping...")
+	} else if err != nil {
+		helpers.PrintError(err)
+		os.Exit(1)
+	} else {
+		raw, err := ioutil.ReadFile(pwd + "/builder.conf")
+		if err != nil {
+			helpers.PrintError(err)
+			os.Exit(1)
+		}
+
+		data := strings.Replace(string(raw), "/home/clr/mix", pwd, -1)
+		data += "\nRPMDIR=" + pwd + "/rpms\n"
+		data += "REPODIR=" + pwd + "/local\n"
+
+		re := regexp.MustCompile("<(.)+>\n")
+		data = re.ReplaceAllString(data, "file://"+pwd+"/update/www\n")
+
+		err = ioutil.WriteFile(pwd+"/builder.conf", []byte(data), 0666)
+	}
+
+	if *confOnlyflag {
+		return
+	}
```

Comment:
Please move the work payload (line 237 through this one) into the builder class. Main.go is currently used for arg parsing and input validation, then passing off to the builder class to do the actual work.

Reply:
It's been in and out of the builder.go file during my revisions. My main argument here is that the work performed is not exclusive related to the builder. But I see an argument to do it otherwise.

Reply:
I see your point, but it's analogous to the work done by builder.InitMix. In general, I am mostly voting for keeping things consistent with how the rest of the code is laid out, for clarity and maintainability.

Reply:
I would like to keep code out of this as well esp if cobra will be replacing the UI here, it will get moved around anyway.